### PR TITLE
Fix bug 1691805: Do not make requests to MT services if locale not supported

### PR DESCRIPTION
**Pull request by mathjazz**
We're getting lots of 400 errors:
https://bugzilla.mozilla.org/show_bug.cgi?id=1691805

### CODE REVIEW

**Comment by mathjazz** [comment]
@Pike r pretty please?

File: pontoon/machinery/views.py [hunk 1]
```diff
@@ -134,7 +134,10 @@ def microsoft_translator(request):
         )
 
     # Validate if locale exists in the database to avoid any potential XSS attacks.
-    if not Locale.objects.filter(ms_translator_code=locale_code).exists():
+    if (
+        not locale_code
+        or not Locale.objects.filter(ms_translator_code=locale_code).exists()
+    ):
```
Comment (Pike) [inline pontoon/machinery/views.py:140]: Now that we only allow these requests for logged-in users, this query lost its purpose. I think we should just remove them.

File: pontoon/machinery/views.py [hunk 2]
```diff
@@ -240,6 +247,8 @@ def systran_translate(request):
             status=404,
         )
 
+    locale = Locale.objects.get(systran_translate_code=locale_code)
```
Comment (Pike) [inline pontoon/machinery/views.py:250]: This could be ambiguous?

**Review by Pike — CHANGES_REQUESTED** [review]
I wish this stuff worked differently, but that's also out of scope, I guess. If we'd eventually chain the MT engines server-side, we'd send the original locale code, and not the custom one. And then bridge that gap on the server side.

I think the frontend changes are OK, but the python parts bother me.

**Comment by mathjazz** [comment]
I've addressed comments in https://github.com/mozilla/pontoon/pull/1862/commits/f654924aa8c08785e18ec31c7dc17eb5ddc3fa33 and https://github.com/mozilla/pontoon/pull/1862/commits/cffb0ea6368a5e070fd01d813c4bd481e05c1147.

In between I've made an additional refactoring.

**Review by Pike — APPROVED** [review]
Thanks, this makes more sense now :-)

**Pull request merged**